### DISCUSSION TITLE
Has anyone gotten an output on a custom voice that's 90% similar?

### DISCUSSION THREAD

**Discussion by vdruts** [neonbjb/tortoise-tts#239] [Polls]
Just wondering. I built a model recently with Talknet and it was quite good! I've been trying to do short clips sentences with more emotion of a similar model with Tortoise but all the output just sounds way too different. I like that the emotionalism really ranges, but I find it alters the pitch/tone of the voice far too much to retain any semblance of likeness.

My input clips are quite good quality, tried 3-5 (3 seems best). So not sure. All 10s.

**Comment by neonbjb** [comment]
Are you looking to get more consistency out of the speech? Try tinkering with the sampling hyperparameters. The AR parameters `temperature` (set it higher, up to 1.0) and `top_p` (set it higher, up to like .95) should help with this.

**Reply by vdruts** [reply]
I'll try that. Just trying to get a high degree of likeness in the voice. The other thing I notified was using prompting like sad or angry made the voice inference come back wildly different than the original, sounding like a young female or crazed.

I think it would be really cool to code a more robust UI to Stable Diffusion or at least explain how the inference works and what settings will have an effect on the model. Steps, Samplers.

Obviously fine tuning and training would be next level.

I'm noticing a truly amazing life like emotion to the voices but indeed they are at this point very different from source material. I'm trying to replicate my voice.

P.S: Where does one set these setting ie) temp?

**Reply by vdruts** [reply]
Perhaps I'm not understanding how to use scripts/tortoise_tts.py as there are only instructions for the python tortoise/do_tts.py --text "I'm going to speak this" --voice random --preset fast

**Comment by neonbjb** [comment]
To get access to these settings you'll need to use the API (or edit `do_tts.py`). They are arguments for the `tts_with_preset` function.

**Reply by vdruts** [reply]
Thanks, figured out the editing of the do_tts. Still a little puzzed at what the settings do. Lower temp seems to provide slower speaking less dramatic but more consistent results. What does top p do exactly
